src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js

```diff
@@ -100,14 +101,48 @@ function preprocessLayer(layer, provider) {
  * @param {Layer} layer.
  */
 ApiGlobe.prototype.addImageryLayer = function(layer) {
-    preprocessLayer(layer, this.scene.managerCommand.getProtocolProvider(layer.protocol));
 
+    preprocessLayer(layer, this.scene.managerCommand.getProtocolProvider(layer.protocol));
     var map = this.scene.getMap();
-
     map.layersConfiguration.addColorLayer(layer);
     this.viewerDiv.dispatchEvent(eventLayerAdded);
 };
 
+/**
+ * This function adds an imagery layer to the scene using a JSON file. The layer id must be unique. The protocol rules wich parameters are then needed for the function.
+ * @constructor
+ * @param {Layer} layer.
+ */
+
+ApiGlobe.prototype.addImageryLayerFromJSON = function(url) {
+     return this.readJSON(url).then(function(result){
```

Comment:
Why not use  `JSONDriver.read(url)` directly instead of adding a one-liner function?